src/service/programService.js

```diff
@@ -171,6 +171,102 @@ function updateProgram(req, response) {
   });
 }
 
+function publishProgram(req, response) {
+  var data = req.body;
+  var rspObj = req.rspObj;
+  if (!data.request || !data.request.program_id) {
+    rspObj.errCode = programMessages.PUBLISH.MISSING_CODE
+    rspObj.errMsg = programMessages.PUBLISH.MISSING_MESSAGE
+    rspObj.responseCode = responseCode.CLIENT_ERROR
+    logger.error({
+      msg: 'Error due to missing request or request program_id',
+      err: {
+        errCode: rspObj.errCode,
+        errMsg: rspObj.errMsg,
+        responseCode: rspObj.responseCode
+      },
+      additionalInfo: {
+        data
+      }
+    }, req)
+    return response.status(400).send(errorResponse(rspObj))
+  }
+
+  model.program.findByPk(data.request.program_id)
+  .then(function (res) {
+    const cb = function(errObj, rspObj) {
+      if (!errObj && rspObj) {
+        const updateValue = {
+          status: "Live",
+          updatedon: new Date(),
+          // copiedCollections: []
+        };
+
+        // _.forEach(rspObj.result, (el, i) => {
+        //   if (el.responseCode === "OK") {
+        //     console.log(el.result.content_id);
+        //     // updateValue.copiedCollections.push(el.result.content_id);
+        //   }
+        // });
+
+        const updateQuery = {
+          where: {
+            program_id: data.request.program_id
+          },
+          returning: true,
+          individualHooks: true,
+        };
+
+        model.program.update(updateValue, updateQuery).then(resData => {
+          if (_.isArray(resData) && !resData[0]) {
+            return response.status(400).send(errorResponse({
+              apiId: 'api.program.publish',
+              ver: '1.0',
+              msgid: uuid(),
+              responseCode: 'ERR_PUBLISH_PROGRAM',
+              result: 'Program_id Not Found'
+            }));
+          }
+          return response.status(200).send(successResponse({
+            apiId: 'api.program.publish',
+            ver: '1.0',
+            msgid: uuid(),
+            responseCode: 'OK',
+            result: {
+              'program_id': updateQuery.where.program_id
+            }
+          }));
+        }).catch(error => {
+          console.log(error)
+          return response.status(400).send(errorResponse({
+            apiId: 'api.program.publish',
+            ver: '1.0',
+            msgid: uuid(),
+            responseCode: 'ERR_PUBLISH_PROGRAM',
+            result: error
+          }));
+        });
+      }
+      else {
+        loggerError(errObj.loggerMsg, errObj.errCode, errObj.errMsg, errObj.responseCode, null, req);
+        return response.status(400).send(errorResponse(errObj));
+      }
+    };
+
+    programServiceHelper.copyCollections(res, 'sunbird', req.headers, cb);
```

Comment:
@ahghatol - get the program by id, if it is present then copy the collection and if collection copy is successful then update the program for status and collection_ids.. This is what need to be done..

Reply:
Also, we already have programCopyCollections function in this file. Why can not we use that instead of writing another one?

Reply:
Yes, did in following already 
1. Get the program id
2. Copy collection
3. Update the program 
I will update the program with collection_ids

we can't use programCopyCollections API, we will be deleting it